### ISSUE TITLE
GAR - Terraform errors with same repo name but different projects

### ISSUE THREAD

**Issue by piotrekkr** [mirakl/terraform-google-gcr-cleaner#58]
### TL;DR

When defining GAR repositories, there is a lot of terraform errors when having same repo name in multiple projects.

### Expected behavior

No terraform errors

### Observed behavior

```
│ Error: Duplicate object key
│ 
│   on .terraform/modules/gcr_cleaner/iam.tf line 44, in resource "google_artifact_registry_repository_iam_member" "this":
│   43:   for_each = {
│   44:     for repo in var.gar_repositories : repo.name => repo
│   45:   }
│     ├────────────────
│     │ repo.name is "api-core"
```

### Terraform Configuration

```hcl
gar_repositories = [
    { region = "europe", project_id = "xxx-staging", name = "api-core" },
    { region = "europe", project_id = "xxx-staging", name = "spooler" },

    { region = "europe", project_id = "xxx-sandbox", name = "api-core" },
    { region = "europe", project_id = "xxx-sandbox", name = "spooler" },

]
```
```


### Terraform Version

```sh
/code $ terraform -version
Terraform v1.2.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.30.0
+ provider registry.terraform.io/hashicorp/google-beta v4.30.0
```

### Additional information

Quick and dirty solutions would be to use something like this in `iam.tf`:
```

# Grant cleaner service account roles/artifactregistry.writer role to read the repository
# and delete container images.
resource "google_artifact_registry_repository_iam_member" "this" {
  for_each = {
    for repo in var.gar_repositories : sha256("${repo.project_id != null ? repo.project_id : local.google_project_id}_${repo.region}_${repo.name}") => repo
  }

```

**Comment by pawelrosada** [comment]
@piotrekkr fixed in `v1.4.1`

**Comment by Erouan50** [comment]
Fixed by #62
